app/gui2/public/visualizations/events.ts

```diff
@@ -0,0 +1,280 @@
+import {
+  computed,
+  onMounted,
+  onUnmounted,
+  proxyRefs,
+  ref,
+  type Ref,
+  shallowRef,
+  watch,
+  watchEffect,
+  type WatchSource,
+} from 'vue'
+
+/**
+ * Add an event listener for the duration of the component's lifetime.
+ * @param target element on which to register the event
+ * @param event name of event to register
+ * @param handler event handler
+ */
+export function useEvent<K extends keyof DocumentEventMap>(
```

Comment:
it is already possible, this is intentionally separate though because this file is largely rewritten to turn `useDocumentEvent` and `useWindowEvent` into overloads rather than separate functions